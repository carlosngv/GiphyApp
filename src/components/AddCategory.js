import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')

    const _handleOnChange = (e) => {
        setinputValue(e.target.value)
    }

    const _handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2) {
            setCategories((c) => [ inputValue, ...c]);
            setinputValue('');
        }
    }

    return (
        <>
        <form onSubmit={_handleSubmit}>
            <input
                type="text"
                className="input is-rounded"
                value={inputValue}
                onChange={_handleOnChange}
                placeholder='Gif'
            />
        </form>
        </>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}