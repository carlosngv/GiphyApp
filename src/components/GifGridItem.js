import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__backInLeft">
            <div className="card-image">
                <figure className="image is-4by3">
                    <img src={url} alt={title} />
                </figure>
            </div>
            <div className="card-content">
                <div className="content">
                    <p>{title}</p>
                </div>
            </div>
        </div>
    )
}


GifGridItem.propTypes =  {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}