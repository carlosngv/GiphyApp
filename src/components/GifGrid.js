import React, { Fragment} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
         
    return (
        <Fragment>
            <h1 className="category-title animate__animated animate__backInRight">{category}</h1>
            <div className="card-grid">
                    {loading && 'Loading...'} 
                    {
                        
                        images.map((img) => (
                            <GifGridItem
                                key={img.id}
                                {...img}
                            />
                        ))
                    }
            </div>
        </Fragment>
    )
}
