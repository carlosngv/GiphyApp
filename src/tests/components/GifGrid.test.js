import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs');

describe('Tests for GifGrid component', () => {

    const testCategory = 'Itachi';

    test('component should work correctyl', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={testCategory} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should fetch images and show them correctly, using the custom hook (useFetchGifs)', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhsot.com/camsdkvsdv.jpg',
            title: 'test'
        }, {
            id: 'DEF',
            url: 'https://localhsot.com/test2.jpg',
            title: 'test2'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={testCategory} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false); // Verifies if loading p tag hides when gifs are fetched
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    });


});