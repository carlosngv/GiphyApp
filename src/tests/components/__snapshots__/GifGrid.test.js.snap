// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGrid component component should work correctyl 1`] = `
<Fragment>
  <h1
    className="category-title animate__animated animate__backInRight"
  >
    Itachi
  </h1>
  <div
    className="card-grid"
  >
    Loading...
  </div>
</Fragment>
`;

exports[`Tests for GifGrid component should fetch images and show them correctly, using the custom hook (useFetchGifs) 1`] = `
<Fragment>
  <h1
    className="category-title animate__animated animate__backInRight"
  >
    Itachi
  </h1>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="test"
      url="https://localhsot.com/camsdkvsdv.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="test2"
      url="https://localhsot.com/test2.jpg"
    />
  </div>
</Fragment>
`;
